src/main/webapp/app/course/competencies/import-standardized-competencies/course-import-standardized-course-competencies.component.ts

```diff
@@ -0,0 +1,190 @@
+import { PrerequisiteService } from 'app/course/competencies/prerequisite.service';
+
+interface StandardizedCompetencyForImport extends StandardizedCompetencyForTree {
+    selected?: boolean;
+    knowledgeAreaTitle?: string;
+}
+
+interface KnowledgeAreaForImport extends KnowledgeAreaForTree {
+    children?: KnowledgeAreaForImport[];
+    competencies?: StandardizedCompetencyForImport[];
+}
+
+@Component({ template: '' })
+export abstract class CourseImportStandardizedCourseCompetenciesComponent extends StandardizedCompetencyFilterPageComponent implements OnInit, ComponentCanDeactivate {
+    protected selectedCompetencies: StandardizedCompetencyForImport[] = [];
+    protected selectedCompetency?: StandardizedCompetencyForImport;
+    protected sourceString = '';
+    protected courseId: number;
+    protected sources: Source[] = [];
+    protected isLoading = false;
+    protected isSubmitted = false;
+
+    // constants
+    protected readonly getIcon = getIcon;
+    protected readonly ButtonType = ButtonType;
+    protected readonly ButtonSize = ButtonSize;
+    readonly documentationType: DocumentationType = 'StandardizedCompetencies';
+    // icons
+    protected readonly faBan = faBan;
+    protected readonly faFileImport = faFileImport;
+    protected readonly faMinimize = faDownLeftAndUpRightToCenter;
+    protected readonly faMaximize = faUpRightAndDownLeftFromCenter;
+    protected readonly faTrash = faTrash;
+    protected readonly faSort = faSort;
+
+    constructor(
+        protected router: Router,
+        protected activatedRoute: ActivatedRoute,
+        protected standardizedCompetencyService: StandardizedCompetencyService,
+        protected alertService: AlertService,
+        protected translateService: TranslateService,
+        protected sortService: SortService,
+    ) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.isLoading = true;
+        const getKnowledgeAreasObservable = this.standardizedCompetencyService.getAllForTreeView();
+        const getSourcesObservable = this.standardizedCompetencyService.getSources();
+        forkJoin([getKnowledgeAreasObservable, getSourcesObservable]).subscribe({
+            next: ([knowledgeAreasResponse, sourcesResponse]) => {
+                const knowledgeAreas = knowledgeAreasResponse.body!;
+                const knowledgeAreasForImport = knowledgeAreas.map((knowledgeArea) => this.convertToKnowledgeAreaForImport(knowledgeArea));
+                this.dataSource.data = knowledgeAreasForImport;
+                this.treeControl.dataNodes = knowledgeAreasForImport;
+                knowledgeAreasForImport.forEach((knowledgeArea) => {
+                    this.addSelfAndDescendantsToMap(knowledgeArea);
+                    this.addSelfAndDescendantsToSelectArray(knowledgeArea);
+                });
+
+                this.sources = sourcesResponse.body!;
+
+                this.isLoading = false;
```

Comment:
the is loading should be moved to the completion block of subscribe otherwise it will stay true if there is an error